src/components/ContactCard/ContactFormField.jsx

```diff
@@ -75,49 +92,90 @@ ContactFieldInput.defaultProps = {
   labelPlaceholder: ""
 };
 
-const ContactFormField = ({
-  icon,
-  name,
-  type,
-  label,
-  placeholder,
-  inputWithLabel,
-  required,
-  labelPlaceholder
-}) => (
-  <div className="contact-form__field">
-    <label className="contact-form__label">
-      {icon && (
-        <Icon icon={icon} color="coolGrey" className="contact-form__icon" />
-      )}
-      {label}
-    </label>
-    <ContactFieldInput
-      name={name}
-      type={type}
-      placeholder={placeholder}
-      withLabel={inputWithLabel}
-      required={required}
-      labelPlaceholder={labelPlaceholder}
-    />
-  </div>
-);
+export class ContactFormField extends React.Component {
+  render() {
+    const { name, icon, label, isArray, renderInput } = this.props;
+    return (
+      <div className="contact-form__field">
+        <label className="contact-form__label">
+          {icon && (
+            <Icon icon={icon} color="coolGrey" className="contact-form__icon" />
+          )}
+          {label}
+        </label>
+        {isArray ? (
+          <FieldArray name={name}>
+            {({ fields }) => (
+              <div className="contact-form__inputs-wrapper">
+                {fields.map((nameWithIndex, index) =>
+                  this.renderArrayField(
+                    fields,
+                    index,
+                    nameWithIndex,
+                    name,
+                    renderInput
+                  )
+                )}
+              </div>
+            )}
+          </FieldArray>
+        ) : (
+          <div className="contact-form__inputs-wrapper">
+            {renderInput(name)}
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  addField = fields => fields.push(undefined);
+
+  removeField = (fields, index) => {
+    const isLastRemainignField = fields.length === 1;
+    fields.remove(index);
+    if (isLastRemainignField) this.addField(fields);
```

Comment:
Yes, it'd be easier to do a `if (fields.length === 0)` after the `remove()`, but `remove()` seems to be async or something, because immediately afterwards `fields.length` is still `1`.